File: src/test/scala/com/kjetland/jackson/jsonSchema/testDataScala/DefaultAndExamples.scala
package com.kjetland.jackson.jsonSchema.testDataScala

import com.fasterxml.jackson.annotation.JsonProperty
import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDefault
import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaExamples;

case class DefaultAndExamples
(
  @JsonSchemaExamples(Array("user@example.com"))
  emailValue:String,
  @JsonSchemaDefault("12")
  @JsonSchemaExamples(Array("10", "14", "18"))
  fontSize:Int,
  @JsonSchemaDefault("2.5")
  @JsonSchemaExamples(Array("1.5", "3.0"))
  weight:Double,

  @JsonProperty( defaultValue = "ds")
  defaultStringViaJsonValue:String,
  @JsonProperty( defaultValue = "1")
  defaultIntViaJsonValue:Int,
  @JsonProperty( defaultValue = "true")
  defaultBoolViaJsonValue:Boolean,
  @JsonProperty( defaultValue = "1.3")
  defaultNumberViaJsonValue:Double
)

File: src/test/java/com/kjetland/jackson/jsonSchema/testData/DefaultAndExamples.java
package com.kjetland.jackson.jsonSchema.testData;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDefault;
import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaExamples;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.experimental.FieldDefaults;
import lombok.extern.jackson.Jacksonized;

@FieldDefaults(level = AccessLevel.PUBLIC, makeFinal = true) @Jacksonized @Builder @EqualsAndHashCode // Can be Java 17 record
public class DefaultAndExamples
{
  @JsonSchemaExamples({"user@example.com"})
  String emailValue;
        
  @JsonSchemaDefault("12")
  @JsonSchemaExamples({"10", "14", "18"})
  int fontSize;

  @JsonProperty(defaultValue = "ds")
  String defaultStringViaJsonValue;

  @JsonProperty(defaultValue = "1")
  int defaultIntViaJsonValue;

  @JsonProperty(defaultValue = "true")
  boolean defaultBoolViaJsonValue;
}